pyaedt/modules/solutions.py

```diff
@@ -2027,111 +1827,104 @@ def polar_plot_3d(
-        if not os.path.exists(geo_path):
-            os.makedirs(geo_path)
-
-        model_pv = self._app.post.get_model_plotter_geometries(plot_air_objects=False)
-
-        obj_meshes = []
-        center = []
-        if is_antenna_array:
-            i = 0
-            for obj in model_pv.objects:
-                for each in data["Element_Location"]:
-                    mesh = obj._cached_polydata
-                    translated_mesh = mesh.copy()
-                    offset_xyz = data["Element_Location"][each] / sf
-                    if np.abs(2 * offset_xyz[0]) > xmax:  # assume array is centere, factor of 2
-                        xmax = offset_xyz[0] * 2
-                    if np.abs(2 * offset_xyz[1]) > ymax:  # assume array is centere, factor of 2
-                        ymax = offset_xyz[1] * 2
-                    translated_mesh.position = offset_xyz
-                    translated_mesh.translate(offset_xyz, inplace=True)
-                    color_cad = [i / 255 for i in obj.color]
-
-                    if len(obj_meshes) > i:
-                        obj_meshes[i][0] += translated_mesh
-                    else:
-                        obj_meshes.append([translated_mesh, color_cad, obj.opacity])
-                i += 1
-                if not center:
-                    center = obj_meshes[-1][0].center
-                else:
-                    center = [i + j for i, j in zip(obj_meshes[-1][0].center, center)]
-        center = [-k / i for k in center]
-        self.all_max = np.max(np.array([xmax, ymax, zmax]))
-        elapsed_time = time.time() - time_before
-        self._app.logger.info("Exporting Geometry...Done: %s seconds", elapsed_time)
-        for mesh in obj_meshes:
-            mesh[0].translate(center, inplace=True)
-        return obj_meshes
+        if show:
+            plot_3d_chart([x, y, z], xlabel="Theta", ylabel="Phi", title=title, snapshot_path=export_image_path)
+        else:
+            return x, y, z
 
+    # fmt: off
     @pyaedt_function_handler()
     def polar_plot_3d_pyvista(
-        self,
-        qty_str="RealizedGain",
-        convert_to_db=True,
-        position=None,
-        rotation=None,
-        export_image_path=None,
-        show=True,
-        show_as_standalone=False,
+            self,
+            farfield_quantity="RealizedGain",
+            format_quantity="dB10",
+            rotation=None,
+            export_image_path=None,
+            show=True,
+            show_as_standalone=False,
+            pyvista_object=None,
+            background=None,
+            scale_farfield=None,
+            show_beam_slider=True,
+            show_geometry=True,
+            **kwargs
     ):
+        # fmt: on
         """Create a 3d Polar Plot of Geometry with Radiation Pattern in Pyvista.
 
         Parameters
         ----------
-        qty_str : str, optional
+        farfield_quantity : str, optional
             Quantity to plot. Default `"RealizedGain"`.
```

Comment:
```suggestion
            Quantity to plot. The default is ``"RealizedGain"``.
```
Add all options as in earlier descriptions.